src/sys/windows/event.rs

```diff
@@ -1,46 +1,41 @@
+use miow::iocp::CompletionStatus;
+
+use crate::sys::windows::afd;
+use crate::Token;
+
 #[derive(Debug)]
 pub struct Event {
     pub flags: u32,
     pub data: u64,
 }
 
-use miow::iocp::CompletionStatus;
-
-use super::afd;
-use crate::Token;
-
 pub fn token(event: &Event) -> Token {
     Token(event.data as usize)
 }
 
 pub fn is_readable(event: &Event) -> bool {
-    if is_error(event) || is_read_hup(event) {
-        return true;
-    }
-    event.flags & (afd::AFD_POLL_RECEIVE | afd::AFD_POLL_ACCEPT) != 0
+    event.flags & (afd::POLL_RECEIVE | afd::POLL_DISCONNECT | afd::POLL_CONNECT | afd::POLL_ACCEPT)
```

Comment:
I don't think ``POLL_CONNECT` should be in here, for two reasons:

* `POLL_CONNECT` indicates that an outbound stream socket has been successfully established. However this does not imply that the server actually sent any data that can be read from the socket. It's very common for this not to be the case (e.g. http servers will not send any data to a client until they've received the request header). Note that select and epoll also don't indicate readability on a connected socket (they will indicate *writability* only).
* The `POLL_CONNECT` flag will go up once a connection has been made, and from there on it *stays up* (remember this is an edge triggered interface). This means that after this patch a connected socket will report that it is *always readable*. I've made this mistake myself in the past (see https://github.com/piscisaureus/wepoll/commit/8b7b340610f88af3d83f40fb728e7b850b090ece).
* Since this bug is apparently not caught by the test, it might be worthwhile to port [this test](https://github.com/piscisaureus/wepoll/blob/d5f8f5f1b1be1a4ba8adb51eb4ee4de7a305a9c8/test/test-connect-success-events.c) to mio.

Reply:
The tried-and-true mappings that wepoll uses: https://github.com/piscisaureus/wepoll/blob/d5f8f5f1b1be1a4ba8adb51eb4ee4de7a305a9c8/src/sock.c#L160-L200

Reply:
I've updated the mappings. Is there are (publicly available) documentation on AFD? I've been searching but couldn't find anything beyond libuv and wepoll.

Reply:
> I've updated the mappings. Is there are (publicly available) documentation on AFD? I've been searching but couldn't find anything beyond libuv and wepoll.

There isn't, but the events listed in the [WSAAsyncSelect](https://docs.microsoft.com/en-us/windows/win32/api/winsock/nf-winsock-wsaasyncselect) docs map 1:1 to `IOCTL_AFD_POLL` events. The only event missing are `AFD_POLL_ABORT` and `AFD_POLL_DISCONNECT`, which are used to indicate that, respectively, a `RST` or `FIN` packet has been received.

Reply:
> I've updated the mappings. Is there are (publicly available) documentation on AFD? I've been searching but couldn't find anything beyond libuv and wepoll.

@Thomasdezeeuw not sure if you found this before: https://github.com/DeDf/afd . Seems to be the afd implementation.